src/server/frontend/render.js

```diff
@@ -99,7 +104,17 @@ function getAppHtml(store, renderProps) {
   );
 }
 
-function getScriptHtml(clientState, headers, hostname) {
+let appAssetFilenameCache = null;
+
+async function getAppAssetFilenamesCachedAsync() { // eslint-disable-line space-before-function-paren
+  if (appAssetFilenameCache) return appAssetFilenameCache;
+
+  appAssetFilenameCache = await getAppAssetFilenamesAsync();
```

Comment:
It's a common pattern also used in C# world where `async/await` was clearly inspired by. 

Moreover, `bluebird` appends `*Async` suffix to functions when `promisify`ing them.